src/zero.ts

```diff
@@ -0,0 +1,56 @@
+import { Response } from "./types";
+import { APIError, APIResultError } from "./errors";
+
+declare const fetch: any; // tslint:disable-line no-any
+
+/**
+ * Zero is a  client to a Dgraph Zero instance.
+ * This is a seperate class because it may be removed in the future.
+ */
+export class DgraphZero {
+    private readonly addr: string;
+
+    /**
+     * Creates a new Client for interacting with Dgraph Zero.
+     *
+     */
+    constructor(addr?: string) {
+        if (addr === undefined) {
+            this.addr = "http://localhost:6080"; // tslint:disable-line no-http-string
+        } else {
+            this.addr = addr;
+        }
+    }
+
+    public getState(): Promise<Response> {
+        return this.callAPI("state", {});
+    }
+
+    getClusterHealth() : Promise<Response> {
+        return this.callAPI("health?all", {});
+    }
+
+    private async callAPI<T>(path: string, config: { method?: string; body?: string; headers?: { [k: string]: string } }): Promise<T> {
+        const url = this.getURL(path);
+
+        return fetch(url, config) // tslint:disable-line no-unsafe-any
```

Comment:
i know it's copy-pasted from elsewhere but can you please refactor it to do `const response = await fetch(...)` and so on. This seems like a good time to introduce modern JS features.

Reply:
Sure